Rename env var prefix MITOPEN_ to MITOL_

rhysyngsun opened the pull request:
### What are the relevant tickets?
<!--- If it fixes an open issue, please link to the issue here. -->
<!--- Closes # --->
<!--- Fixes # --->
<!--- N/A --->
Part of https://github.com/mitodl/hq/issues/5154

### Description (What does it do?)
<!--- Describe your changes in detail -->
Renames the environment variables to the more non product specific `MITOL_`.

Temporary code added to `manage.py` to use the legacy prefix if variables are present.

### How can this be tested?
<!---
Please describe in detail how your changes have been tested.
Include details of your testing environment, any set-up required
(e.g. data entry required for validation) and the tests you ran to
see how your change affects other areas of the code, etc.
Please also include instructions for how your reviewer can validate your changes.
--->
- Checkout the first commit and verify you can `docker compose up` without errors 
- Checkout the second commit and verify the same

Ardiea commented on .github/workflows/ci.yml:
```diff
@@ -123,7 +123,7 @@ jobs:
         run: yarn run build
         env:
-          MITOPEN_API_BASE_URL: https://api.mitopen-test.odl.mit.edu
+          MITOL_API_BASE_URL: https://api.mitopen-test.odl.mit.edu
```
Do we want to update these URLs while we're here or is that too many moving parts at the moment?

rhysyngsun replied:
It's a dummy value but I can update it.

Ardiea replied:
Well this one, yes. But in the production and rc files they are real. Thinking about it maybe they should come from ol-inf with much of the other config.

rhysyngsun replied:
That's going to depend on the timing, if I update those, the next build that goes out the frontend will try to connect to those immediately and things will break hard if they're not completely in place. I think maybe the answer is that they should come from a secret, not from a static value.

rhysyngsun replied:
Actually I'm inclined to say that changing the url values is out of scope for this issue. I'll leave the one I already touched but the others should get dealt with separately because otherwise it's comingling two different kinds of changes (variable name and value changes).

Ardiea replied:
Sounds good!

rhysyngsun commented:
@Ardiea I addressed your feedback in that one line you commented, but I think anything more is out of scope as noted [here](https://github.com/mitodl/mit-open/pull/1388#discussion_r1710229408).

blarghmatey reviewed (approved):
This all looks good from a semantic perspective but I'll defer to someone else for some functional testing.

Ardiea commented on .github/workflows/release-candidate.yml:
```diff
@@ -44,10 +44,10 @@ jobs:
-          MITOPEN_SUPPORT_EMAIL: mitlearn-support@mit.edu
+          SENTRY_ENV: ${{ secrets.MITOL_ENV_RC }}
+          MITOL_AXIOS_WITH_CREDENTIALS: true
+          MITOL_API_BASE_URL: https://api.mitopen-rc.odl.mit.edu
```
Okay, with the infrastructure where it is now, this vault SHOULD be changed at this point.
